src/main/java/co/rsk/federate/config/PowHSMBookkeepingConfig.java

```diff
@@ -1,87 +1,86 @@
-    public static final BigInteger DIFFICULTY_CAP_MAINNET = new BigInteger("7000000000000000000000");
-    public static final BigInteger DIFFICULTY_CAP_TESTNET = BigInteger.valueOf(1000000000000000L);
-    public static final BigInteger DIFFICULTY_CAP_REGTEST = BigInteger.valueOf(20);
-    private final BigInteger difficultyCap;
-    private BigInteger difficultyTarget = BigInteger.valueOf(3);
-    private int maxAmountBlockHeaders = 7;
-    private long informerInterval = 2_000;
-    private boolean stopBookkeepingScheduler = false;
-    private int maxChunkSizeToHsm = 10;
 
-    public PowHSMBookkeepingConfig(SignerConfig signerConfig, String networkParameter) {
-        if (signerConfig.getConfig().hasPath("bookkeeping.difficultyTarget")) {
-            this.difficultyTarget = new BigInteger(signerConfig.getConfig().getString("bookkeeping.difficultyTarget"));
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.maxAmountBlockHeaders")) {
-            this.maxAmountBlockHeaders = signerConfig.getConfig().getInt("bookkeeping.maxAmountBlockHeaders");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.informerInterval")) {
-            this.informerInterval = signerConfig.getConfig().getLong("bookkeeping.informerInterval");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.stopBookkeepingScheduler")) {
-            this.stopBookkeepingScheduler = signerConfig.getConfig().getBoolean("bookkeeping.stopBookkeepingScheduler");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.maxChunkSizeToHsm")) {
-            this.maxChunkSizeToHsm = signerConfig.getConfig().getInt("bookkeeping.maxChunkSizeToHsm");
+    public enum NetworkDifficultyCap {
+        MAINNET(new BigInteger("7000000000000000000000")),
+        TESTNET(BigInteger.valueOf(1000000000000000L)),
+        REGTEST(BigInteger.valueOf(20L));
+
+        private final BigInteger difficultyCap;
+
+        NetworkDifficultyCap(BigInteger difficultyCap) {
+            this.difficultyCap = difficultyCap;
         }
 
-        switch (networkParameter) {
-            case NetworkParameters.ID_MAINNET:
-                this.difficultyCap = DIFFICULTY_CAP_MAINNET;
-                break;
-            case NetworkParameters.ID_TESTNET:
-                this.difficultyCap = DIFFICULTY_CAP_TESTNET;
-                break;
-            case NetworkParameters.ID_REGTEST:
-                this.difficultyCap = DIFFICULTY_CAP_REGTEST;
-                break;
-            default:
-                String message = "Invalid network specified for the Bookkeeping Config: " + networkParameter;
-                logger.error(message);
-                throw new IllegalArgumentException(message);
+        public BigInteger getDifficultyCap() {
+            return difficultyCap;
         }
     }
 
+    private static final BigInteger DEFAULT_DIFFICULTY_TARGET = BigInteger.valueOf(3);
```

Comment:
This should be gone, right? Is that part of upcoming changes?

Reply:
Yep, this will be done in the next PR. This is just a refactor before introducing "new" changes, so it easier to review.